lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1395,6 +1395,33 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     return nullptr;
   }
 
+  template <typename DeclType> bool isFilePrivate(DeclType *D) {
+    return (D->hasAccess() && D->getFormalAccess() == AccessLevel::FilePrivate);
```

Comment:
Hm. This isn't correct because `AccessLevel::Private` also qualifies, but if you're really going to be this precise then it's probably worth factoring out a common check with ASTMangler's `getPrivateDiscriminatorIfNecessary`. Otherwise this will almost certainly drift in the future.